### ISSUE TITLE
Is the TCP implementation still not optimized?

### ISSUE THREAD

**Issue by freak82** [OpenFastPath/ofp#232]
Hi there,

In this page - https://openfastpath.org/index.php/service/technicaloverview/ I read that
```
IP and UDP implementations have been optimized for performance, TCP implementation is functional but not performance optimized.
```

Is this still the case?
Is the TCP implementation ready to be used against traffic in the wild? I mean, all kinds of HTTP and peer to peer traffic from different OS that an Internet Service Provider sees, for example.

Thanks,
Pavel.

**Comment by bogdanPricope** [comment]
Yes and No. Many optimizations were tried (mostly in private forks). Thing is, there is not a single solution that fits all cases. What may work on "webserver" kind of scenarios may not work on single connection with large throughput. 

About "ready" and "traffic in the wild".. afaik no guarantees are provided.

**Comment by freak82** [comment]
Hi,

Thanks for the response.

I was asking about the following scenario: transparent forwarding proxy with caching capabilites used by some internet service providers for bandwidth saving. We have multiple locations where the  proxy processes between 100 MBps and 20 Gbps. The peers on both sides of the proxy connection can use literally everything as OS and network stack implementation and we have no control over it.
We are searching for ways to decrease our hardware requirements using technologies like OpenFastPath or F-stack (dpdk + FreeBSD networking stack).

Pavel.
